[Bug] geosx_xml_tools and Xcode builds

**Describe the bug**
I am attempting to configure an Xcode build as follows
```
python scripts/config-build.py -hc host-configs/joshua.cmake -bt Debug --xcode
```
This leads to a configuration error:
```
CMake Error in coreComponents/python/CMakeLists.txt:
  The custom command generating

    /Users/white230/Documents/code/geosx/GEOSX/build-joshua-debug/bin/preprocess_xml

  is attached to multiple targets:

    geosx_xml_tools_test
    geosx_format_all_xml_files
    geosx_xml_tools

  but none of these is a common dependency of the other(s).  This is not
  allowed by the Xcode "new build system".
```
I modified the CMakeLists.txt file as follows, which seems to resolve the issue.  I replaced `DEPENDS ${GEOSX_XML_TOOLS_BINS}` with `DEPENDS geosx_xml_tools` in a few spots. I don't know if this is a good patch though.  @cssherman?
```
add_custom_command( OUTPUT ${GEOSX_XML_TOOLS_BINS}
                    COMMAND bash ${CMAKE_SOURCE_DIR}/../scripts/setupVirtualPythonEnvironment.bash -p ${PYTHON_POST_EXECUTABLE} -o ${CMAKE_BINARY_DIR}/python
                    COMMAND ln -s ${CMAKE_BINARY_DIR}/python/geosx/bin/preprocess_xml ${CMAKE_BINARY_DIR}/bin/preprocess_xml
                    COMMAND ln -s ${CMAKE_BINARY_DIR}/python/geosx/bin/format_xml ${CMAKE_BINARY_DIR}/bin/format_xml
                    COMMAND ln -s ${CMAKE_BINARY_DIR}/python/geosx/bin/python ${CMAKE_BINARY_DIR}/bin/python
                    WORKING_DIRECTORY ${CMAKE_BINARY_DIR}
                  )

add_custom_target( geosx_xml_tools
                   DEPENDS ${GEOSX_XML_TOOLS_BINS} )

add_custom_target( geosx_xml_tools_test
                   COMMAND ${CMAKE_BINARY_DIR}/python/geosx/bin/test_geosx_xml_tools
                   COMMAND rm -r ${CMAKE_BINARY_DIR}/python/geosx_xml_tools_tests*
                   WORKING_DIRECTORY ${CMAKE_BINARY_DIR}/python
                   DEPENDS geosx_xml_tools
                 )

add_custom_target( geosx_format_all_xml_files
                   COMMAND bash ${CMAKE_SOURCE_DIR}/../scripts/formatXMLFiles.bash -g ${CMAKE_BINARY_DIR}/bin/format_xml ${CMAKE_SOURCE_DIR} ${CMAKE_SOURCE_DIR}/../examples
                   WORKING_DIRECTORY ${CMAKE_BINARY_DIR}
                   DEPENDS geosx_xml_tools
                 )
```

--- a/src/coreComponents/python/CMakeLists.txt
+++ b/src/coreComponents/python/CMakeLists.txt
@@ -43,12 +43,12 @@ add_custom_target( geosx_xml_tools_test
                    COMMAND ${CMAKE_BINARY_DIR}/python/geosx/bin/test_geosx_xml_tools
                    COMMAND rm -r ${CMAKE_BINARY_DIR}/python/geosx_xml_tools_tests*
                    WORKING_DIRECTORY ${CMAKE_BINARY_DIR}/python
-                   DEPENDS ${GEOSX_XML_TOOLS_BINS}
+                   DEPENDS geosx_xml_tools 
                  )
 
 add_custom_target( geosx_format_all_xml_files
                    COMMAND bash ${CMAKE_SOURCE_DIR}/../scripts/formatXMLFiles.bash -g ${CMAKE_BINARY_DIR}/bin/format_xml ${CMAKE_SOURCE_DIR} ${CMAKE_SOURCE_DIR}/../examples
                    WORKING_DIRECTORY ${CMAKE_BINARY_DIR}
-                   DEPENDS ${GEOSX_XML_TOOLS_BINS}
+                   DEPENDS geosx_xml_tools 
                  )
 

